packages/devtools_app/lib/src/shared/console/widgets/display_provider.dart

```diff
@@ -56,30 +56,42 @@ class DisplayProvider extends StatelessWidget {
     }
 
     final hasName = variable.name?.isNotEmpty ?? false;
+
+    // The tooltip can be hovered over in order to see the original text.
+    final originalDisplayValue = variable.displayValue.toString();
+    // Only 1 line of text is permitted in the tree, since we only get 1 row.
+    // So replace all newlines with \\n so that the user can still see that
+    // there are new lines in the value.
+    final displayValue = originalDisplayValue.replaceAll('\n', '\\n');
     return InteractivityWrapper(
       onTap: onTap,
       menuButtons: _getMenuButtons(context),
-      child: Text.rich(
-        TextSpan(
-          text: hasName ? variable.name : null,
-          style: variable.artificialName
-              ? theme.subtleFixedFontStyle
-              : theme.fixedFontStyle.apply(
-                  color: theme.colorScheme.controlFlowSyntaxColor,
+      child: DevToolsTooltip(
+        message: variable.displayValue.toString(),
+        child: Text.rich(
+          maxLines: 1,
+          overflow: TextOverflow.ellipsis,
```

Comment:
instead of showing the \n chars, can we just make this a multiline text widget? With some reasonable number of max lines like 3 or 4? We would still ellipsize and have a tooltip.